### PR TITLE
feat: Added support for CQA with unit tests.

### PR DESCRIPTION
**Pull request by Arsh-Kashyap**
#minor

## Description
Added Language Service utils for querying Language Service via bot. DefaultAnswer can be configured from secrets file. Updated unit tests and added tests for CQA(Custom Question Answering).

## How to test
1) Use [this](https://github.com/microsoft/BotBuilder-Samples/pull/3716) PR for samples.
2) Run `yarn` and `yarn build` in [root folder](https://github.com/Arsh-Kashyap/botbuilder-js/tree/Arsh-Kashyap/LanguageService) of SDK.
3) Run `yarn build` in [botbuilder-ai](https://github.com/Arsh-Kashyap/botbuilder-js/tree/Arsh-Kashyap/LanguageService/libraries/botbuilder-ai) and [botbuilder-dialogs](https://github.com/Arsh-Kashyap/botbuilder-js/tree/Arsh-Kashyap/LanguageService/libraries/botbuilder-dialogs) in SDK.
4) Run `yarn link` in [botbuilder-dialogs](https://github.com/Arsh-Kashyap/botbuilder-js/tree/Arsh-Kashyap/LanguageService/libraries/botbuilder-dialogs) in sdk.
5) Run `yarn link botbuilder-dialogs` in [botbuilder-ai](https://github.com/Arsh-Kashyap/botbuilder-js/tree/Arsh-Kashyap/LanguageService/libraries/botbuilder-ai). Then run `yarn link`. 
6) Go to samples and run `npm install` , `yarn link botbuilder-ai` and `yarn link botbuilder-dialogs` in [48.customQABot-all-features](https://github.com/Arsh-Kashyap/BotBuilder-Samples/tree/Arsh/NodejsCQA/samples/javascript_nodejs/48.customQABot-all-features).
7) Run `npm start` to test.
8) Same steps can be performed to test [12.customQABot](https://github.com/Arsh-Kashyap/BotBuilder-Samples/tree/Arsh/NodejsCQA/samples/javascript_nodejs/12.customQABot), and previous samples  [11.qnamaker](https://github.com/Arsh-Kashyap/BotBuilder-Samples/tree/Arsh/NodejsCQA/samples/javascript_nodejs/11.qnamaker) and  [49.qnamaker-all-features](https://github.com/Arsh-Kashyap/BotBuilder-Samples/tree/Arsh/NodejsCQA/samples/javascript_nodejs/49.qnamaker-all-features).

## Specific Changes
<li> Added Language Service utils for querying Language Service via bot.
<li> Default Answer can now be configured from .env file in nodejs sample.
<li> If Default Answer is not mentioned it will render the one filled in language studio.

## Testing
Added Language Service tests and updated  existing tests.

### CODE REVIEW

**Review by mrivera-ms — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-dialogs/package.json [hunk 1]
```diff
@@ -31,9 +31,9 @@
     "@microsoft/recognizers-text-date-time": "1.1.4",
     "@microsoft/recognizers-text-number": "1.1.4",
     "@microsoft/recognizers-text-suite": "1.1.4",
-    "botbuilder-core": "4.1.6",
-    "botbuilder-dialogs-adaptive-runtime-core": "4.1.6",
-    "botframework-connector": "4.1.6",
+    "botbuilder-core": "4.14.1",
+    "botbuilder-dialogs-adaptive-runtime-core": "4.14.1-preview",
+    "botframework-connector": "4.14.1",
```
Comment (mrivera-ms) [inline libraries/botbuilder-dialogs/package.json:36]: Why is this needed?

File: libraries/botbuilder-testing/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     }
   },
   "dependencies": {
-    "botbuilder-core": "4.1.6",
+    "botbuilder-core": "4.14.1",
```
Comment (mrivera-ms) [inline libraries/botbuilder-testing/package.json:30]: Why this?

**Review by mrivera-ms — CHANGES_REQUESTED** [review]

**Comment by mrivera-ms** [comment]
@Arsh-Kashyap, please add code coverage to `qnaCardBuilder.ts`, `generateAnswerUtil.ts` and `languageServiceUtil.ts`. Particularly qnaCardBuilder.ts which is dropping to 38.71%:

### Before:
image

### After:
image

File: libraries/botbuilder-ai/package.json [hunk 3]
```diff
@@ -29,11 +29,11 @@
   "dependencies": {
     "@azure/cognitiveservices-luis-runtime": "2.0.0",
     "@azure/ms-rest-js": "1.9.1",
-    "adaptive-expressions": "4.1.6",
-    "botbuilder-core": "4.1.6",
-    "botbuilder-dialogs": "4.1.6",
-    "botbuilder-dialogs-adaptive-runtime-core": "4.1.6",
-    "botbuilder-dialogs-declarative": "4.1.6",
+    "adaptive-expressions": "4.9.0",
+    "botbuilder-core": "4.14.1",
```
[the remark this answers is not in the document]
Comment (Arsh-Kashyap) [reply libraries/botbuilder-ai/package.json:33]: [Adaptive Expressions](https://www.npmjs.com/package/adaptive-expressions) 4.1.6 is not supported. Along with that botbuilder-dialogs-adaptive-runtime-core and botbuilder-dialogs-declarative is also not supporting 4.1.6. Let me know if some other team is handling these version changes. I can revert them from my side, but updating them would be required to run the SDK.

File: libraries/botbuilder-ai/src/qnamaker-interfaces/qnamakerOptions.ts [hunk 4]
```diff
@@ -71,4 +73,14 @@ export interface QnAMakerOptions {
      * A value indicating choice for Strict Filters Join Operation.
      */
     strictFiltersJoinOperator?: JoinOperator;
+
+    /**
+     * A value indicating user's choice to receive PreciseAnswer or not.
+     */
+    enablePreciseAnswer?: boolean;
```
[the remark this answers is not in the document]
Comment (Arsh-Kashyap) [reply libraries/botbuilder-ai/src/qnamaker-interfaces/qnamakerOptions.ts:80]: I have set a default value in the constructor of CustomQuestionAnswering.ts. Is there a way to set default value in the interface itself?

File: libraries/botbuilder-dialogs/src/skillDialog.ts [hunk 5]
```diff
@@ -335,7 +335,7 @@ export class SkillDialog extends Dialog<Partial<BeginSkillDialogOptions>> {
             (c) => c.contentType === CardFactory.contentTypes.oauthCard
         );
         if (oAuthCardAttachment) {
-            const tokenExchangeProvider: ExtendedUserTokenProvider = context.adapter as ExtendedUserTokenProvider;
+            const tokenExchangeProvider: ExtendedUserTokenProvider = (context.adapter as unknown) as ExtendedUserTokenProvider;
```
[the remark this answers is not in the document]
Comment (Arsh-Kashyap) [reply libraries/botbuilder-dialogs/src/skillDialog.ts:338]: Without "as unknown", It gives an error. I think working of some referred packages have changed
image

**Review by mrivera-ms — APPROVED** [review]
